prom/poller.go

```diff
@@ -203,11 +220,28 @@ func suffixFor(m *dto.Metric) string {
 	result := make([]string, 0, len(m.Label))
 
 	for _, lp := range m.Label {
-		result = append(result, lp.GetName()+"_"+lp.GetValue())
+		labelName := strings.Map(charMapper, lp.GetName())
+		labelVal := strings.Map(charMapper, lp.GetValue())
+		result = append(result, labelName+"_"+labelVal)
 	}
 
 	if len(result) == 0 {
 		return ""
 	}
 	return "." + strings.Join(result, ".")
 }
+
+func charMapper(r rune) rune {
```

Comment:
To me, this makes more sense since there is really only `return r` or `return '_'`. 

```go
func charMapper(r rune) rune {
	if r >= 'a' && r <= 'z' || r >= 'A' && r <= 'Z' ||
		r >= '0' && r <= '9' || r == '-' || r == '.' {
		return r
	}
	return '_'
}
```

or maybe

```go
func charMapper(r rune) rune {
	switch {
	case r >= 'a' && r <= 'z' || r >= 'A' && r <= 'Z' ||
		r >= '0' && r <= '9' || r == '-' || r == '.':
		return r
	default:
		return '_'
	}
}
```